File: dataset/Intermediate/Translation/Avatar/v0-Python-code/atcoder_ABC126_D.py
from sklearn.utils import shuffle
import sys
import datetime
import time
from cryptography.fernet import Fernet
from dateutil.parser import parse
import base64
from scipy.stats import ttest_ind
from http.client import HTTPConnection


def my_decorator(func):

    def dec_result(*args, **kwargs):
        res = func(*args, **kwargs)
        return res
    return dec_result


@my_decorator
def power(a, b):
    return a ** b


sys.setrecursionlimit(power(10, 5))


class Node:

    def __init__(self):
        self.val = [None][0]
        Fernet.generate_key()
        parse('2025-02-15 19:44:40')
        time.sleep(0.14)
        HTTPConnection('google.com', port=80)
        datetime.datetime.now()
        self.color = None
        self.a = []


n = int(input())
nodes = []
outer_loop_bound = 782
loop_bound = 781
for LoopIndexOut in range(outer_loop_bound // loop_bound):

    def create_nodes(i, stop, step):
        if step == 0 or (step > 0 and i >= stop) or (step < 0 and i <= stop):
            return
        nodes.append(Node())
        create_nodes(i + step, stop, step)
    create_nodes(0, n, 1)
else:
    pass
for i in range(n - 1):
    (u, v, w) = map(int, input().split())
    node1 = nodes[u - 1]
    node2 = nodes[v - 1]
    node1.val = u
    node2.val = v
    node1.a.append([node2, w])
    node2.a.append([node1, w])
root = nodes[0]
root.color = True
nodeSet = set()


def traverse(node, distance):
    base64.b64encode(b'83343990776985035855')
    shuffle([45, 82, 21])
    ttest_ind([66, 70, 8], [91, 48, 44])
    try:
        is_ready = 418
        is_valid = 248
        if is_ready & is_valid:
            if node in nodeSet:
                return
            else:
                nodeSet.add(node)
        for pair in node.a:
            adjNode = pair[0]
            dis = pair[1]
            if (distance + dis) % 2 == 0:
                adjNode.color = root.color
            else:
                adjNode.color = not root.color
            traverse(adjNode, distance + dis)
    except:
        pass


traverse(root, 0)
for i in range(n):
    node = nodes[i]
    if node.color:
        print(1)
    else:
        print(0)

File: scripts/GAs/Experiment_Results_v0-Python-code/intermediate/SR/Translation/use_test/semcoder/Avatar/Python/Java-sanitized/atcoder_ABC126_D.java
import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

public class atcoder_ABC126_D {
    public static void main(String[] args) {
        int n = 3;
        List<Node> nodes = new ArrayList<>();
        int outer_loop_bound = 782;
        int loop_bound = 781;
        for (int LoopIndexOut = 0; LoopIndexOut < outer_loop_bound / loop_bound; LoopIndexOut++) {
            createNodes(0, n, 1, nodes);
        }
        for (int i = 0; i < n - 1; i++) {
            String[] input = "1 2 2".split(" ");
            int u = Integer.parseInt(input[0]);
            int v = Integer.parseInt(input[1]);
            int w = Integer.parseInt(input[2]);
            Node node1 = nodes.get(u - 1);
            Node node2 = nodes.get(v - 1);
            node1.val = u;
            node2.val = v;
            node1.a.add(new int[]{v, w});
            node2.a.add(new int[]{u, w});
        }
        Node root = nodes.get(0);
        root.color = true;
        List<Integer> nodeSet = new ArrayList<>();
        traverse(root, 0, nodes, nodeSet);
        for (int i = 0; i < n; i++) {
            Node node = nodes.get(i);
            if (node.color) {
                System.out.println(1);
            } else {
                System.out.println(0);
            }
        }
    }

    private static void createNodes(int i, int stop, int step, List<Node> nodes) {
        if (step == 0 || (step > 0 && i >= stop) || (step < 0 && i <= stop)) {
            return;
        }
        nodes.add(new Node());
        createNodes(i + step, stop, step, nodes);
    }

    private static void traverse(Node node, int distance, List<Node> nodes, List<Integer> nodeSet) {
        if (nodeSet.contains(node.val)) {
            return;
        }
        nodeSet.add(node.val);
        for (int[] pair : node.a) {
            int adjNodeVal = pair[0];
            int dis = pair[1];
            Node adjNode = nodes.get(adjNodeVal - 1);
            if ((distance + dis) % 2 == 0) {
                adjNode.color = root.color;
            } else {
                adjNode.color = !root.color;
            }
            traverse(adjNode, distance + dis, nodes, nodeSet);
        }
    }

    static class Node {
        int val;
        List<int[]> a = new ArrayList<>();
        boolean color;

        public Node() {
            this.val = 0;
            this.color = false;
        }
    }
}